src/pocketmine/plugin/PluginManager.php

```diff
@@ -158,6 +158,7 @@ public function loadPlugin(string $path, array $loaders = null) : ?Plugin{
 				if($description instanceof PluginDescription){
 					$this->server->getLogger()->info($this->server->getLanguage()->translateString("pocketmine.plugin.load", [$description->getFullName()]));
 					try{
+						$description->checkCompatibleOperatingSystem();
 						$description->checkRequiredExtensions();
```

Comment:
`checkRequiredExtensions()` was already misplaced here; your check should more likely be placed in [this](https://github.com/pmmp/PocketMine-MP/pull/3192/files#diff-10c30a902a542d91471a1a004b245c13R256-R286) block, following the style of other compatibility checks.